core/schema/module.go

```diff
@@ -321,182 +380,1162 @@ func (s *moduleSchema) functionWithArg(ctx context.Context, fn *core.Function, a
 	return fn.WithArg(args.Name, argType.Self, args.Description, args.DefaultValue), nil
 }
 
-func (s *moduleSchema) moduleWithSource(ctx context.Context, self *core.Module, args struct {
-	Directory core.DirectoryID
-	Subpath   string `default:""`
+func (s *moduleSchema) moduleDependency(
+	ctx context.Context,
+	query *core.Query,
+	args struct {
+		Source core.ModuleSourceID
+		Name   string `default:""`
+	},
+) (*core.ModuleDependency, error) {
+	src, err := args.Source.Load(ctx, s.dag)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode dependency source: %w", err)
+	}
+	src.Self = src.Self.Clone()
```

Comment:
Is this needed? Seems like it's anticipating modifying the `Instance` value directly which is generally a no-no

Reply:
No, was leftover while I was debugging/questioning my sanity due to a bug I had previously where I was overwriting `ModuleDependency` values that were cached (fixed [here](https://github.com/dagger/dagger/pull/6386/commits/8fdb548ea2c19fd06caf21ec53bb64584ba0abe0)). Will remove now